File: 1770-minimum-deletions-to-make-character-frequencies-unique/minimum-deletions-to-make-character-frequencies-unique.cs
public class Solution {
    public int MinDeletions(string s) {
        int[] freq = new int[26];
        
        // Count frequency of each character
        foreach (char c in s) {
            freq[c - 'a']++;
        }

        var usedFrequencies = new HashSet<int>();
        int deletions = 0;

        foreach (int f in freq) {
            int currFreq = f;
            while (currFreq > 0 && !usedFrequencies.Add(currFreq)) {
                // Keep reducing until the frequency is unique
                currFreq--;
                deletions++;
            }
        }

        return deletions;
    }
}

File: 1770-minimum-deletions-to-make-character-frequencies-unique/minimum-deletions-to-make-character-frequencies-unique.cpp
class Solution {
public:
    int minDeletions(string s) {
        vector<int> freq(26, 0);

        for(char c : s)
        {
            freq[c - 'a']++;
        }

        unordered_set<int> usedFreq;
        int deletions = 0;

        for(int f : freq)
        {
            int currentFreq = f;
            while(currentFreq > 0 && (usedFreq.count(currentFreq) > 0))
            {
                currentFreq--;
                deletions++;
            }
            usedFreq.insert(currentFreq);
        }

        return deletions;
    }
};